### PR TITLE
Warm cold

### PR DESCRIPTION
**Pull request by LeonPev**

### CODE REVIEW

File: internal/transform/span.go [hunk 1]
```diff
@@ -103,6 +103,7 @@ func (m *mapper) Transform(invocationStartedTimestamp int64) telemetry.Span {
 		isWarmStart := os.Getenv("IS_WARM_START")
 		if isWarmStart == "" && !isProvisionConcurrencyInitialization() {
 			lumigoSpan.LambdaReadiness = "cold"
+			os.Setenv("IS_WARM_START", "true")
```
Comment (nirLumigo) [inline internal/transform/span.go:106]: Did you check to verify that it works?
I think that the function `Transform` is called for both the start-span and end-span. If this is the case then I think that in cold start for the start-span we'll have `lumigoSpan.LambdaReadiness = "cold"` but for the end-span we'll have `lumigoSpan.LambdaReadiness = "warm"`

**Review by nirLumigo — COMMENTED** [review]
Nice :)

File: internal/transform/span.go (hunk 1 above)
Comment (LeonPev) [reply internal/transform/span.go]: you are right. fixed

File: wrapper_test.go [hunk 2]
```diff
@@ -282,6 +285,11 @@ func (w *wrapperTestSuite) TestLambdaHandlerE2ELocal() {
 			assert.Equal(w.T(), "bd862e3fe1be46a994272793", startFuncSpan.TransactionID)
 			assert.Equal(w.T(), string(inputPayload), startFuncSpan.Event)
 			assert.Equal(w.T(), version, startFuncSpan.SpanInfo.TracerVersion.Version)
+			if i > 0 { // check if second test
+				assert.Equal(w.T(), "warm", startFuncSpan.LambdaReadiness)
+			} else {
+				assert.Equal(w.T(), "cold", startFuncSpan.LambdaReadiness)
+			}
```
Comment (nirLumigo) [inline wrapper_test.go:292]: I think we should have a variable `expectedReadiness` so it won't break easily. Same regarding the test of the endSpan.
With the current implementation, if we'll change the order of the tests then the test will break.

**Review by nirLumigo — CHANGES_REQUESTED** [review]

**Review by nirLumigo — APPROVED** [review]

**Pull request merged**